channel/state/outcome/outcome.go

```diff
@@ -55,6 +56,32 @@ func (a Allocations) Total() *big.Int {
 	return total
 }
 
+// Affords returns true if the allocations can afford the given allocation given input funds of x, false otherwise.
+//
+// To afford the given allocation, the allocations must include something equal-in-value to it,
+// as well as having sufficient funds left over for it after reserving funds from x for all allocations with higher priority.
+// Note that "equal-in-value" implies the same allocation type and metadata (if any).
+func (allocations Allocations) Affords(given Allocation, x *big.Int) bool {
+	bigZero := big.NewInt(0)
+	surplus := big.NewInt(0).Set(x)
+	for _, allocation := range allocations {
+
+		if surplus.Cmp(bigZero) == 0 {
+			break
+		}
+
+		affords := math.BigMin(surplus, allocation.Amount)
+
+		if allocation.Equal(given) && affords.Cmp(allocation.Amount) >= 0 {
+			return true
+		}
+
+		surplus.Sub(surplus, affords)
+
+	}
+	return false
+}
```

Comment:
- moves the most important logical check to the top of the loop
  - ... and returns a result directly, whether T or F
- a little shorter - removes accounting step
- adds comment on the `break`
- ... passes tests

```suggestion
func (allocations Allocations) Affords(given Allocation, x *big.Int) bool {
	bigZero := big.NewInt(0)
	surplus := big.NewInt(0).Set(x)
	for _, allocation := range allocations {

		if allocation.Equal(given) {
			return surplus.Cmp(given.Amount) >= 0
		}

		surplus.Sub(surplus, allocation.Amount)

		if surplus.Cmp(bigZero) != 1 {
			break // no funds remain for further allocations
		}

	}
	return false
}
```